src/lib.ts

```diff
@@ -1,15 +1,33 @@
+// ----- Imports ----- //
+
+import { ReactElement } from 'react';
+import { Option, map, withDefault } from '@guardian/types/option';
+
+
 // ----- Functions ----- //
 
 const pipe = <A, B>(a: A, f: (_a: A) => B): B => f(a);
 const pipe2 = <A, B, C>(a: A, f: (_a: A) => B, g: (_b: B) => C): C => g(f(a));
 const pipe3 = <A, B, C, D>(a: A, f: (_a: A) => B, g: (_b: B) => C, h: (_c: C) => D): D =>
     h(g(f(a)));
 
+function MaybeRender<A>(props: {
```

Comment:
That's an interesting implementation of `map` right here 😄

Reply:
Yeah I did notice that in the type signature. Fundamentally JSX is just an alternate syntax for making function calls. This is why I'm sometimes reluctant to use it, because it has a tendency to wrap what are otherwise quite simple operations in a layer of mis-direction.